src/actions/getFavoriteRecipes.ts

```diff
@@ -0,0 +1,19 @@
+import { prisma } from "../lib/prisma";
+
+const getFavoriteRecipes = async (userId: string | undefined) => {
+  const favoriteRecipes = await prisma.favorite.findMany({
+    where: {
+      userId,
+    },
+    include: {
+      recipe: true,
+    },
+    orderBy: {
+      createdAt: "desc",
+    },
+  });
+
+  return favoriteRecipes;
+};
```

Comment:
`getAuthenticatedUser`で認証済みのユーザーid取れるので引数でuserId渡す必要ないと思いました！

```suggestion
const getFavoriteRecipes = async () => {
  const user = await getAuthenticatedUser();

  if (!user) {
    return [];
  }

  const favoriteRecipes = await prisma.recipe.findMany({
    where: {
      userId: user.id,
    },
    include: {
      likes: true,
    },
    orderBy: {
      createdAt: "desc",
    },
  });

  return favoriteRecipes;
};
```

Reply:
@tuboihirokidesu 
ありがとうございます。指摘いただいたコードが良いと思ったので修正します。

※業務であれば以下のようにしたいと思いましたので、共有してさせてください。
頂いた書き方だとgetFavoriteRecipesの処理がユーザの存在に依存するので避けたい書き方だと思いました。
つまり副作用をもつ関数になります。今回のプロジェクトだと関係ないかもですが、テストコードが書きづらいコードになると思いました。
また、取得結果が空配列の場合ユーザ認証が通っていないから空配列なのかお気に入りが存在しないから空配列なのか判定できないように見えます。

Reply:
> 取得結果が空配列の場合ユーザ認証が通っていないから空配列なのかお気に入りが存在しないから空配列なのか判定できないように見えます

たしかにそうですね！
認証が失敗した場合とお気に入りが存在しない場合で返す値を区別するために、認証が失敗した場合はnullを返すようにして、

```ts
const getFavoriteRecipes = async () => {
  const user = await getAuthenticatedUser();
  if (!user) {
    return null; 
  }

  ...

    return favoriteRecipes;
}
```

呼び出し側でnullの場合はエラー返すようにすればいいと思いました！

```tsx
const page = async () => {
  const favoriteRecipes = await getFavoriteRecipes();

  if (favoriteRecipes === null) return notFound();

  console.log(favoriteRecipes);

  return <div>kameのmock page</div>;
};
```

> 頂いた書き方だとgetFavoriteRecipesの処理がユーザの存在に依存するので避けたい書き方だと思いました。
つまり副作用をもつ関数になります

んー、自分の意見を言うと
以下の処理の時点でユーザーが認証されていない場合にリダイレクトを行うという追加の動作を持っているため、この動作は、関数がアプリケーションの状態（具体的にはユーザーの認証状態）を変更する可能性があるため、副作用と見なされると思います！


```ts
const supabaseServerClient = createServerComponentClient<Database>({ cookies });

const {
  data: { session },
} = await supabaseServerClient.auth.getSession();

if (!session) {
  redirect("/mock/unauthenticated");
}
```

なのでログイン済みのuserIdを必要とする場合、SC(Server Component)で上記の処理を書いて引数でuserId渡すのは冗長で、getAuthenticatedUserで認証済みのuserId取るのが良いと思いレビュさせてもらいました！